### PR TITLE
Introduce `#[structopt(external_subcommand)]`

### LINKED ISSUE #130: Use of AllowExternalSubcommands
I wonder if it's possible to use [external subcommands](https://docs.rs/clap/2.32.0/clap/enum.AppSettings.html#variant.AllowExternalSubcommands).

I tried adding some `raw(setting = "structopt::clap::AppSettings::AllowExternalSubcommands")` here and there, but `Opt::from_args()` always shows the help then exits.

### PR DESCRIPTION
**Pull request by CreepySkeleton**
Closes #130 

I took some time to develop and debug, but it's finally ready. I have one concern thought.

### CODE REVIEW

File: tests/subcommands.rs [hunk 1]
```diff
@@ -211,3 +205,93 @@ fn flatten_enum() {
...
+        Opt::from_iter(&["test", "bar", "--flag", "--arg=0"]),
+        Opt {
+            sub_cmd: SubCmd::Other(vec!["bar".into(), "--flag".into(), "--arg=0".into()])
+        }
+    );
+    assert_eq!(
+        Opt::from_iter(&["test"]),
+        Opt {
+            sub_cmd: SubCmd::Other(vec!["".into()])
+        }
```
Comment (CreepySkeleton) [inline tests/subcommands.rs]: What's the behavior should be in this case? We could pass `SubcommandRequiredOrHelp` along with `AllowExternalSubcommands`

Comment (TeXitoi) [reply tests/subcommands.rs]: Agree, without option on the subcommand, the subcommand must be provided, and that's not the case here.

**Comment by CreepySkeleton** [comment]
Sorry for the radio silence, the New Year and all :)

This is not abandoned, I'll get to it in 1-2 days.

File: tests/ui/external_subcommand_wrong_type.stderr [hunk 2]
```diff
@@ -0,0 +1,8 @@
+error[E0308]: mismatched types
+  --> $DIR/external_subcommand_wrong_type.rs:13:15
+   |
+13 |     Other(Vec<CString>)
+   |               ^^^^^^^ expected struct `std::ffi::CString`, found struct `std::ffi::OsString`
+   |
+   = note: expected type `std::vec::Vec<std::ffi::CString>`
+              found type `std::vec::Vec<std::ffi::OsString>`
```
Comment (CreepySkeleton) [inline tests/ui/external_subcommand_wrong_type.stderr]: This error message is rather unfortunate: I would very much like it to say "expected OsString", but there's nothing we can do about it 😢

**Comment by CreepySkeleton** [comment]
OK, ready to go

File: tests/ui/external_subcommand_wrong_type.stderr (hunk 2 above)
Comment (CreepySkeleton) [reply tests/ui/external_subcommand_wrong_type.stderr]: This is the best I've managed to come up with:

```
error[E0308]: mismatched types
  --> $DIR/external_subcommand_wrong_type.rs:13:15
   |
13 |     Other(Vec<CString>)
   |               ^^^^^^^ expected struct `std::ffi::CString`, found struct `std::ffi::OsString`
   |
   = note: expected type `std::vec::Vec<std::ffi::CString>`
              found type `std::vec::Vec<std::ffi::OsString>`

error[E0308]: mismatched types
  --> $DIR/external_subcommand_wrong_type.rs:13:15
   |
13 |     Other(Vec<CString>)
   |               ^^^^^^^
   |               |
   |               expected struct `std::ffi::OsString`, found struct `std::ffi::CString`
   |               this match expression has type `std::vec::Vec<std::ffi::OsString>`
   |
   = note: expected type `std::vec::Vec<std::ffi::OsString>`
              found type `std::vec::Vec<std::ffi::CString>`

```

While it is indeed possible to draw the error message we desire, I can't get rid of the first one.

**Comment by CreepySkeleton** [comment]
ping @TeXitoi

File: tests/ui/external_subcommand_wrong_type.stderr (hunk 2 above)
Comment (TeXitoi) [reply tests/ui/external_subcommand_wrong_type.stderr]: That's acceptable

**Pull request merged**